components/collator/src/tests.rs

```diff
@@ -69,13 +70,15 @@ fn check_expectations(
 
 #[test]
 fn test_basic() {
-    let data_provider = icu_testdata::get_provider();
-
     let mut options = CollatorOptions::new();
     options.set_strength(Some(Strength::Quaternary));
 
-    let collator: Collator =
-        Collator::try_new_unstable(&data_provider, &Default::default(), options).unwrap();
+    let collator: Collator = Collator::try_new_unstable(
+        &icu_testdata::buffer().as_deserializing(),
```

Comment:
Thought: In general, I'd rather cover `try_new_with_buffer_provider` than use `try_new_unstable` with `as_deserializing()`

(No action required)

Reply:
I moved this to an integration test